src/v/kafka/server/consumer_group_lag_metrics_frontend.cc

```diff
@@ -124,15 +124,19 @@ consumer_group_lag_metrics_frontend::get_partition_offsets(
 ss::future<partition_offsets_reply>
 consumer_group_lag_metrics_frontend::get_local_partition_offsets(
   partition_offsets_request req) {
-    auto ktps = request_as_ktps(req);
+    // const because don't share mutable state between threads.
+    const auto ktps = request_as_ktps(req);
 
     auto gate_holder = _gate.hold();
 
     // gather all cores
     co_return co_await container().map_reduce0(
-      [&ktps](auto& me) {
+      [ktps](auto& me) {
+          // take a copy of ktps since it is an input_range, which requires it
```

Comment:
yikes foot gun

Reply:
> yikes foot gun

Yeah, it was a bit tricky.

Fortunately, if you make the range const (as you should if you're sharing the state), the compiler does do a fair amount of complaining.